x/stakeibc/keeper/icqcallbacks_withdrawal_balance_test.go

```diff
@@ -228,6 +228,22 @@ func (s *KeeperTestSuite) TestWithdrawalBalanceCallback_NoFeeAccount() {
 	s.Require().EqualError(err, "no fee account found for GAIA: ICA acccount not found on host zone")
 }
 
+func (s *KeeperTestSuite) TestWithdrawalBalanceCallback_FailedToCheckForRebate() {
```

Comment:
Note to self @riley-stride -  review this again once you understand the difference between `WithdrawalBalanceCallback` and `WithdrawalRewardBalanceCallback`

Reply:
I see Ethan already addressed it, but difference is that `WithdrawalBalanceCallback` queries for native tokens, and `WithdrawalRewardBalanceCallback` queries for USDC (or any other revenue token that's been registered with the trade route)

The reason their handled differently is because in `WithdrawalRewardBalanceCallback`, we need the amount to trade (reinvestAmount + strideFee) and the rebateAmount.

Whereas in `WithdrawalBalanceCallback`, we're post-trade and post-rebate, so we only need strideFee and reinvestAmount.

Reply:
Ty!

Leaving open to until we address [this](https://github.com/Stride-Labs/stride/pull/1136#issuecomment-2002047681)